zio-config-refined/src/main/scala/zio/config/refined/internal/TypeInference.scala

```diff
@@ -0,0 +1,187 @@
+package zio.config.refined
```

Comment:
Let's change the name of TypeInference to `TypeInferenceSupport.scala`
We should also make this private as well.

Reply:
Or even PariallyAppliedTypes.scala

The word `TypeInference` is too generic.

Reply:
Agreed